src/overlays/actors/ovl_En_Egol/z_en_egol.c

```diff
@@ -119,99 +320,1319 @@ const ActorInit En_Egol_InitVars = {
+}
+
+void EnEgol_StunEnd(EnEgol* this, PlayState* play) {
+    f32 frame = this->skelAnime.curFrame;
+
+    if (frame >= this->animEndFrame) {
+        EnEgol_SetupWalk(this);
+    }
+}
+
+void EnEgol_SetupSlamEnd(EnEgol* this) {
+    EnEgol_SetAnimation(this, EYEGORE_ANIM_SLAM_END);
+    this->actionTimer = 0;
+    this->actionState = EYEGORE_ACTION_SLAM_END;
+    this->actionFunc = EnEgol_SlamEnd;
+}
+
+void EnEgol_SlamEnd(EnEgol* this, PlayState* play) {
+    f32 frame = this->skelAnime.curFrame;
+
+    if (frame >= this->animEndFrame) {
+        EnEgol_SetupWalk(this);
+    }
+}
+
+void EnEgol_SetupDamaged(EnEgol* this) {
+    EnEgol_SetAnimation(this, EYEGORE_ANIM_DAMAGED);
+    this->bodyCollider.elements[0].dim.modelSphere.radius = 20;
+    this->bodyCollider.elements[1].dim.modelSphere.radius = 20;
+    this->actionState = EYEGORE_ACTION_DAMAGED;
+    this->actionFunc = EnEgol_Damaged;
+}
+
+void EnEgol_Damaged(EnEgol* this, PlayState* play) {
+    f32 frame = this->skelAnime.curFrame;
 
-extern ColliderJntSphElementInit D_80A80C64[6];
-extern ColliderJntSphInit D_80A80D3C;
-extern ColliderJntSphElementInit D_80A80C40[1];
-extern ColliderJntSphInit D_80A80D4C;
-extern ColliderQuadInit D_80A80D5C;
-extern DamageTable D_80A80DAC;
+    if (frame >= this->animEndFrame) {
+        if (this->actor.colChkInfo.health > 0) {
+            EnEgol_SetupWalk(this);
+        } else {
+            Enemy_StartFinishingBlow(play, &this->actor);
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_EYEGOLE_DEAD);
+            this->actor.flags |= ACTOR_FLAG_8000000;
+            this->actor.flags &= ~ACTOR_FLAG_1;
+            this->actor.flags |= ACTOR_FLAG_100000;
+            this->actionFunc = EnEgol_StartDeathCs;
+        }
+    }
+}
 
-extern UNK_TYPE D_06000040;
-extern UNK_TYPE D_060094E4;
-extern UNK_TYPE D_0600EE4C;
+void EnEgol_StartDeathCs(EnEgol* this, PlayState* play) {
+    if (!ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    } else {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        this->subCamId = ActorCutscene_GetCurrentSubCamId(this->actor.cutscene);
+        this->subCamFovTarget = 60.0f;
+        EnEgol_SetAnimation(this, EYEGORE_ANIM_DEATH);
+        this->actionState = EYEGORE_ACTION_DYING;
+        this->actionFunc = EnEgol_Death;
+        this->actor.shape.shadowScale = 0.0f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7C990.s")
+void EnEgol_Death(EnEgol* this, PlayState* play) {
+    f32 curFrame = this->skelAnime.curFrame;
+    Vec3f atOffset;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Egol/func_80A7CA18.s")
+    this->subCamAt.x = 0.0f;
+    this->subCamAt.y = 60.0f;
+    this->subCamAt.z = 260.0f;
+    Math_Vec3f_Copy(&atOffset, &this->subCamAt);
+    OLib_DbCameraVec3fSum(&this->actor.world, &atOffset, &this->subCamAt, 1);
```

Comment:
Not a comment for this PR, but a self-noted comment that this function name needs to eventually change. Doesn't need `DbCamera` in it